torch/csrc/autograd/profiler.h

```diff
@@ -203,6 +203,65 @@ TORCH_API void mark(std::string name, bool include_cuda = true);
 TORCH_API void pushRange(std::string name);
 TORCH_API void popRange();
 
+struct TORCH_API FunctionCallContext {
+  FunctionCallContext(Function* fn);
+  FunctionCallContext(std::string name, int64_t sequence_nr = -1);
+  FunctionCallContext(const char* name_ptr, int64_t sequence_nr = -1);
+
+  Function* func() const {
+    return fn_;
+  }
+
+  const char* name() const {
+    return name_ptr_;
+  }
+
+  int64_t seq_nr() const {
+    return sequence_nr_;
+  }
+
+ private:
+  Function* fn_ = nullptr;
+  std::string name_;
```

Comment:
Having to keep and initialize an `std::string` per event might be a bit much (especially that this is a rarely used functionality). That's why we used to keep `std::unique_ptr<std::string>` an an Event (there's a comment explaining that too).